File: code/java/DesignPatterns/src/main/java/command/Waiter.java
package command;

class Waiter {
    private Command command;

    void setCommand(Command command) {
        this.command = command;
    }

    void executeCommand() {
        System.out.println("Waiter.executeCommand(): Delegating to Command.execute() for " + command.getClass().getSimpleName());
        command.execute();
    }
}

File: code/csharp/DesignPatterns/command/Waiter.cs
namespace Command;

class Waiter
{
    private ICommand _command;

    public ICommand Command
    {
        get => _command;
        set
        {
            _command = value;
            System.Console.WriteLine($"Waiter setting command to {value
                                                            .GetType().Name}");
        }
    }

    public void ExecuteCommand()
    {
        System.Console.WriteLine($"Waiter.ExecuteCommand(): " +
            "Delegating to Command.Execute() for " + _command.GetType().Name);

        _command.Execute();
    }
}
